### ISSUE TITLE
Instant Crash

### ISSUE THREAD

**Issue by codechimera** [nkzawa/socket.io-client.java#165]
Hello there!

I'm trying to implement the client as in the example on your blog post, but I'm using a dedicated class for the connection:

````java
 private static Socket mSocket;
    {
        try {
            mSocket = IO.socket(Constants.SOCKET_URI);
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
        }
    }

    public static void connect() {
        if (mSocket.connected()) { return; }

        Log.d(TAG, "Connecting to socket IO server");
        mSocket.connect();
        mSocket.on("connected", onSocketConnected);
    }

    private static Emitter.Listener onSocketConnected = new Emitter.Listener() {
        @Override
        public void call(Object... args) {
            Log.d(TAG, "Socket connected with ID: " + mSocket.id());
        }
    };
````

However when I run .connect(), my app immediately crashes. 
Any ideas? I don't see any error messages in my ADB logs.

**Comment by nkzawa** [comment]
Your code looks good to me. We need logs or more details.

btw, there is no "connected" event, mistake of "connect" (Socket.EVENT_CONNECT)?

**Comment by nabeelpkl** [comment]
is this code running in UI thread?

**Comment by shuoli84** [comment]
@codechimera still has the problem?

**Comment by codechimera** [comment]
I replaced the connect event as @nkzawa suggested, and I found that the issue was that I was not running the code inside an activity; apparently if I put the same exact code inside an activity (instead of a "Server" class as I had been) it runs fine. 

I am not sure why this would be, but I suppose it works- it just seems somewhat silly to have to dis/reconnect to the server every time I move to a new activity. I am extremely new to Android development though so this could easily be something I messed up somewhere.

If that explanation indicates something I'm doing wrong I'd love to hear it, but otherwise this can be closed! Thanks for the assistance everyone :)
